docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,108 @@
+  
+  Rel(pebble, charm, "")
+}
+```
+
+### Jenkins
+
+This container runs the main workload of the charm. The OCI image is custom built and includes
+the [Jenkins WAR](https://www.jenkins.io/doc/book/installing/war-file/), Java and the [Jenkins plugin installation manager](https://github.com/jenkinsci/plugin-installation-manager-tool/).
+
+To facilitate monitoring of the Jenkins application via the COS, the [prometheus](https://plugins.jenkins.io/prometheus/) plugin is installed by default. The [instance-identity](https://plugins.jenkins.io/instance-identity/) plugin is also included to ensure seamless registration of agent nodes. Additionally, the [monitoring](https://plugins.jenkins.io/monitoring/) plugin is pre-installed to handle session invalidation.
+
+When a logging relation is joined, a promtail application is started via Pebble which starts
+pushing Jenkins logs at `/var/lib/jenkins/logs/jenkins.log` to Loki.
+The metrics are also scraped via accessing the `/metrics` endpoint of the Jenkins application.
+
+### charm
+
+This container is the main point of contact with the Juju controller. It communicates with Juju to
+run necessary charm code defined by the main `src/charm.py`. The source code is copied to the
+`/var/lib/juju/agents/unit-UNIT_NAME/charm` directory.
+
+## OCI image
+
+The jenkins-image is custom built to include Jenkins as well as its plugin installation manager. Since Jenkins is
+an application running on Java, required libraries and dependencies are installed during the build
+process.
+
+Jenkins application installation is done at runtime during container pebble ready step.
+
+Currently, Jenkins version 2.492.2 is used alongside Ubuntu 22.04 base image.
+
+## Integrations
+
+See [Relation endpoints](../reference/integrations.md).
+
+### Peer relations
+
+Only one deployment per Jenkins application is supported.
+
+## Juju events
+
+Juju events allow progression of the charm in its lifecycle and encapsulates part of the execution
+context of a charm. Below is the list of observed events for `jenkins-k8s charm` with how the charm
+reacts to the event. For more information about the charm’s lifecycle in general, refer to the
+charm’s life [documentation](https://canonical-juju.readthedocs-hosted.com/en/3.6/user/reference/hook/).
+
+### jenkins_pebble_ready
+
+This event signals that the Pebble inside the workload container is ready. The charm then starts interacting with Pebble to begin the installation process.
+
+### _on_jenkins_home_storage_attached
+
+This event marks the charm’s storage availability. The name of the event derived from the name of
+the storage noted in the `metadata.yaml` configuration under "storage" key.
+`containers.jenkins.mounts.storage` and `storage.jenkins-home` section. The storage filesystem maps to
+`/var/lib/jenkins` directory of the Jenkins application, which is used to store Jenkins related files.
+
+### _on_update_status
```

Comment:
Any particular reason these two headings include the `_on` when others don't please?

Reply:
Good catch, updated, thank you :)